rsts/community/contribute.rst

```diff
@@ -284,68 +284,373 @@ To understand how the below components interact with each other, refer to :ref:`
     * - **Guidelines**: Refer to the `FlyteCTL Contribution Guide <https://docs.flyte.org/projects/flytectl/en/stable/contribute.html>`__    
 
 
-🔮 Recommended iteration cycle
-==============================
-
-As you may have already read in other parts of the documentation, the `Flyte repository <https://github.com/flyteorg/flyte>`__ includes Go code
-that integrates all backend components (admin, propeller, data catalog, console) into a single executable.
-The Flyte team is currently working on consolidating the core backend repositories into one repository, which is expected to be completed by 2023.
-In the meantime, you can contribute to the individual repositories and then merge your changes into the `Flyte repository <https://github.com/flyteorg/flyte>`__.
-This setup is suitable for Go-based backend development, but it has not been tested for Flyteconsole development, which has a different development cycle.
-Nonetheless, this setup allows you to run the Flyte binary from your IDE, enabling you to debug your code effectively by setting breakpoints.
-Additionally, this setup connects you to all other resources in the demo environment, such as PostgreSQL and RDS.
-
-Dev mode cluster
-****************
+🔮 Development Environment Setup Guide
+======================================
+
+This guide provides a step-by-step approach to setting up a local development environment for `flyteidl <https://github.com/flyteorg/flyteidl>`_, `flyteadmin <https://github.com/flyteorg/flyteadmin>`_, `flyteplugins <https://github.com/flyteorg/flyteplugins>`_, `flytepropeller <https://github.com/flyteorg/flytepropeller>`_, `flytekit <https://github.com/flyteorg/flytekit>`_ and `flyteconsole <https://github.com/flyteorg/flyteconsole>`_.
+
+Requirements
+************
+
+This guide has been tested and used on AWS EC2 with an Ubuntu 22.04
+image. The following tools are required:
+
+- `Docker <https://docs.docker.com/install/>`__
+- `Kubectl <https://kubernetes.io/docs/tasks/tools/install-kubectl/>`__
+- `Go <https://golang.org/doc/install>`__
+
+Content
+*******
+
+-  `How to set dev environment for flyteidl, flyteadmin, flyteplugins,
```

Comment:
"set" -> "setup"?

same comment for following lines.

Reply:
also, not sure how much we want to list all the components - this would also include flytestdlib, datacatalog, etc i think